packages/payload/src/utilities/configToJSONSchema.ts

```diff
@@ -93,6 +93,41 @@ function generateEntitySelectSchemas(
   }
 }
 
+function generateCollectionJoinsSchemas(collections: SanitizedCollectionConfig[]): JSONSchema4 {
+  const properties = [...collections].reduce<Record<string, JSONSchema4>>(
```

Comment:
I don't think we would want to have all collections added to `collectionJoins` do we? I would expect this to be filtered down to only the few that have joins defined.

Reply:
Updated and now it's even better, if the collection doesn't have any joins the result type for `joins` is `never`, therefore you can't use it.

<img width="730" alt="image" src="https://github.com/user-attachments/assets/c58176ed-6e79-4f43-aafd-09d85cd89105">